src/library/rekordbox/rekordboxfeature.cpp

```diff
@@ -0,0 +1,932 @@
+                << "currentPath" << currentPath;
+            return;
+        }        
+
+        QSqlQuery idQuery(m_database);
+        idQuery.prepare("select id from rekordbox_playlists where name=:path");
+        idQuery.bindValue(":path", currentPath);
+
+        if (!idQuery.exec()) {
+            LOG_FAILED_QUERY(idQuery)
+                << "currentPath" << currentPath;
+            return;
+        }        
+
+        int playlistID = -1;
+        while (idQuery.next()) {
+            playlistID = idQuery.value(idQuery.record().indexOf("id")).toInt();
+        }        
+
+        QSqlQuery queryInsertIntoPlaylistTracks(m_database);
+        queryInsertIntoPlaylistTracks.prepare(
+            "INSERT INTO rekordbox_playlist_tracks (playlist_id, track_id, position) "
+            "VALUES (:playlist_id, :track_id, :position)");        
+
+        if (playlistTrackMap.count(childID)) {
+            // Add playlist tracks for children
+            for (uint32_t trackIndex = 1; trackIndex <= playlistTrackMap[childID].size(); trackIndex++) {
+                uint32_t rbTrackID = playlistTrackMap[childID][trackIndex];
+
+                int trackID = -1;
+                QSqlQuery finderQuery(m_database);
+                finderQuery.prepare("select id from rekordbox_library where rb_id=:rb_id and device=:device");
+                finderQuery.bindValue(":rb_id", rbTrackID);
+                finderQuery.bindValue(":device", device);
+
+                if (!finderQuery.exec()) {
+                    LOG_FAILED_QUERY(finderQuery) 
+                        << "rbTrackID:" << rbTrackID
+                        << "device:" << device;
+                    return;
+                }
+
+                if (finderQuery.next()) {
+                    trackID = finderQuery.value(finderQuery.record().indexOf("id")).toInt();
+                }
+
+                queryInsertIntoPlaylistTracks.bindValue(":playlist_id", playlistID);
+                queryInsertIntoPlaylistTracks.bindValue(":track_id", trackID);
+                queryInsertIntoPlaylistTracks.bindValue(":position", (int)trackIndex);
+
+                if (!queryInsertIntoPlaylistTracks.exec()) {
+                    LOG_FAILED_QUERY(queryInsertIntoPlaylistTracks)
+                        << "playlistID:" << playlistID
+                        << "trackID:" << trackID
+                        << "trackIndex:" << trackIndex;
+
+                    return;
+                }                              
+            }
+        }
+
+        if (playlistIsFolderMap[childID]) {
+            // If this child is a folder (playlists are only leaf nodes), build playlist tree for it
+            buildPlaylistTree(child, childID, playlistNameMap, playlistIsFolderMap, 
+                playlistTreeMap, playlistTrackMap, currentPath, device);
+        }
+    }
+}                                                
+
+void RekordboxFeature::clearTable(QString tableName) {
```

Comment:
By passing the database connection as an argument you can transform this member function into a simple utility function, hidden in the .cpp file.